### DISCUSSION TITLE
Disconnecting a relation during an update() causes database to be written to twice

### DISCUSSION THREAD

**Discussion by tophboogie** [prisma/prisma#20672] [Q&A]
### Question

Hello,
I am not sure if this is expected behavior or if I'm doing something incorrect...

On a prisma.update() call (a soft delete), I would like to disconnect a one-to-one related record if it exists. This causes the database to be written to twice instead of just once. 

The first write it changes deleted to true, then on the second write it changes the sessionId column to null. This wouldn't be a big deal except that we have triggers set to log database changes and we are seeing duplicate entries because of this. Any ideas? Thanks!!
```js
return await database.workOrder.update({
  where: { id: workOrder.id },
  data: {
    deleted: true,
    session: workOrder.sessionId ? { disconnect: true } : undefined,
    updatedBy: { connect: { id: userId } }
  }
});
```
### Expected behavior (optional)

I expected the database to be updated once, showing just one updated row in the logs

### Information about Prisma Schema, Client Queries and Environment (optional)
```prisma
model Session {
  id              Int        @id @default(autoincrement())
  ...
  workOrder       WorkOrder?
  updatedAt       DateTime   @updatedAt
  updatedBy       User       @relation(fields: updatedByUserId, references: id)
  updatedByUserId Int
  deleted         Boolean    @default(false)
}

model WorkOrder {
  id              Int      @id @default(autoincrement())
  ...
  session         Session? @relation(fields: sessionId, references: id)
  sessionId       Int?     @unique
  updatedAt       DateTime @updatedAt
  updatedBy       User     @relation(fields: updatedByUserId, references: id)
  updatedByUserId Int
  deleted         Boolean  @default(false)
}
```
- OS: Ubuntu 20.04
- Database: MySQL
- Node.js version: v18.16.1
- Prisma version: 4.16.1

**Comment by ludralph** [comment]
Hi @tophboogie 👋 

The behavior you're observing is likely due to how Prisma handles updates with multiple fields, especially when there are cascading actions or interrelated fields involved. In your case, you want to update the `deleted` field and disconnect the `session` field if it exists. The issue arises from the fact that Prisma is performing these actions in two separate database operations, resulting in the duplicate log entries.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by tophboogie** [comment]
Hi @ludralph 
Thank you for the explanation. Is there any way to accomplish this with just one database operation? We have many other prisma calls that update multiple fields but this is the first we've seen multiple database operations. 

Is there a reason this call requires multiple operations? Or is there some documentation as to when multiple operations are required? 

Unfortunately, the duplicate log entries is a big deal for our application so I'll need to dig into this more and find a way around it. Any advice or insight you have would be greatly appreciated, thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
